pkg/mcproducer/scripts/replace-kernel-module.sh

```diff
@@ -19,17 +21,36 @@ EOF
     cat "$kmm_config_file_filepath"
 }
 
-echo "before checking image tar file presence"
-if [ -e $kernel_module_image_filepath ]; then
-    echo "Image file $kernel_module_image_filepath found on the local file system, creating kmm config file"
+echo "before checking image presence"
+if [ -n "$(podman images -q $full_kernel_module_image 2> /dev/null)" ]; then
+    echo "Image $full_kernel_module_image found on the local file system, creating kmm config file"
     create_kmm_config
-    echo "running kernel-management worker image"
-    podman run --user=root --privileged -v /lib/modules:/lib/modules -v $kmm_config_file_filepath:/etc/kmm-worker/config.yaml -v $kernel_module_image_filepath:$kernel_module_image_filepath $worker_image kmod load --tarball /etc/kmm-worker/config.yaml
+    mkdir -p /tmp/opt/lib/modules
```

Comment:
I would use `podman volume create` instead of manually creating a directory on the host.
https://docs.podman.io/en/latest/markdown/podman-volume-create.1.html